### DISCUSSION TITLE
Current status

### DISCUSSION THREAD

**Discussion by ste-gr** [ste-gr/struts1-reloaded#14] [General]
Dear JEE9-Struts1-interested,

since it seems that this project is getting more and more interest, I would like to share with the current status and I will also update it if there are any changes.

And here it is :smile::
* I have started to change the namespaces from `javax` to `jakarta`.
* Unfortunately, the following dependencies are not yet JEE9+ compatible:
   * [commons-fileupload](https://github.com/apache/commons-fileupload) (there is now version 2 which is JEE9+ compatible).
      * Open Tasks: Adapt Struts1 to use the new version. **DONE**
   * [commons-chain](https://github.com/weblegacy/commons-chain) (I'm working on this right now; the current apache-version is very old :disappointed:)
      * Tasks:
         * Update to current MAVEN-Plugins-Versions **DONE**
         * Add MAVEN-Plugins for static analyses (PMD/CPD, Checkstyle, ...) **DONE**
         * Generics **DONE**
         * JPMS **DONE**
         * Split it to submodules and add jakarta-versions **DONE**
         * Update docs **DONE**
         * Release it **DONE**
   * [tiles-request-servlet](https://github.com/weblegacy/tiles-request-servlet) (I still want to have a look at the current solution, write a small doc and then publish it)
      * Tasks:
         * Add MAVEN-Plugins for static analyses (PMD/CPD, Checkstyle, ...) **DONE**
         * Write a short doc/README **DONE**
         * JPMS **DONE**
         * OSGi **DONE**
         * Release it **DONE**

Nice to have, but not necessary:
* [tlddoc](https://github.com/weblegacy/tlddoc) - Tag Library Documentation Generator
   * Update to taglib 3.0 (from 2.1) **WiP**
   * [Testcases](https://github.com/weblegacy/tlddoc/issues/1)
* [taglib-maven-plugin](https://github.com/weblegacy/taglib-maven-plugin) - Update to new tlddoc

Any contribution is welcome!

Greetings
Stefan

**Comment by lukamalovic** [comment]
Hi Stefan, 

I've worked on updating to commons-fileupload2 locally. Would I be able to get added as a contributor so that I can push my changes to a feature branch and have them be reviewed?

Thanks,
Luka

**Reply by ste-gr** [reply]
Hi Luka,

great, thank you. I sent you the invitation to add you as member to this repo.

Greetings
Stefan

**Comment by kubera** [comment]
Hello Stefan, how is the plan to continue with the Java version? We are using Struts 1 on java 11 and should upgrade to Java 17.

Would it help if I have a look what has to be done? If I contribute, would you accept a pull request on java 17 for a upcomming version?

Regards, Stefan W.

**Reply by ste-gr** [reply]
Hello Stefan,

every contribution is welcome! After the 1.5.0 version (JakartaEE 9+) I would like to adapt to JakartaEE 10/JDK-11 and use `module-info.java` instead of `Automatic-Module-Name`. And adapt to JDK-17 (or other JDK) with multi-release jar files.

Which JEE environment are you using?

Best regards
Stefan

**Reply by kubera** [reply]
Thanks for the fast reply!

We migrated two Websphere 8.5 apps EJB2/Java8 into a Spring 5 and Kubernet environment on Java 11. JEE is not important for us anymore but the namespace change from javax to jakarta hurts.

We should upgrade to Spring 6 with Tomcat 10 and Java 17. Currently  that's not possible with Struts 1.

**Reply by ste-gr** [reply]
Hello Stefan,

then the 1.5.0 version is just right for you :smile:. Another member has tested the current 1.5.0 version with Wildfly 30 without any problems.

Greetings
Stefan

**Reply by kubera** [reply]
Oh, that's even better. Thanks for the info.

I'll let you know, when I have some first tests..

**Reply by anabright** [reply]
Hi @ste-gr,

I wanted to thank you for all your effort and time invested in making Struts 1 compatible with the jakarta namespace.
Our app uses Spring + Struts 1 and when I tried to upgrade it to Spring 6 we ran into the hurdle of Struts 1 being incompatible. I've been testing it with the 1.5.0-SNAPSHOT version and it seems to work fine.
I ran into a few other issues, for example Freemarker not being JEE9+ compatible but I've managed to fork the repo and create a compatible version.

@kubera I've used Spring 6 + Struts 1 + Java17 throughout all my tests and I haven't found any issues so far

Great work!

**Reply by ste-gr** [reply]
Hi @anabright,

thank you for your tests and report! If you like, we can also publish your other work on weblegacy.

Thank you and
Greetings
Stefan

**Reply by anabright** [reply]
No problem, I will let you know once we're done upgrading our app to see which repos/artifacts we can publish here.

Hopefully by then Freemarker will have released a new version with support for jakarta namespace, see here https://github.com/apache/freemarker/pull/94#issuecomment-1858791651

Are there any plans to publish v1.5.0 in maven-repo soon?

Ana

**Reply by kubera** [reply]
Do you see a release date for the verson 1.5.0 ? 
I have the impression that all is marked as done or nice to have.

**Reply by ste-gr** [reply]
Hi @anabright, @kubera!

Sorry for the late replay! I'll publish version `1.5.0-RC1` the the next few days, because `tlddoc` is not able to generate the taglib-docs (upgrade to Jakarta Server Pages 3.0 is missing, I'm working on it).
The only different between `1.5.0-RC1` and `1.5.0` will be the taglib docs. The version `1.5.0-RC1` can therefore be used without further ado.

Greetings
Stefan

**Reply by ste-gr** [reply]
Hi @anabright, @kubera!

Great news: The Struts1-Version **ready for Jakarta-Namespace** (JEE9+) has been released (Version 1.5.0-RC1)!

Greetings
Stefan

**Reply by anabright** [reply]
Thanks Stefan! Will it be published on maven at some point?

Ana

**Reply by ste-gr** [reply]
Hi Ana,

it is on maven: [Struts Core - 1.5.0-RC1](https://mvnrepository.com/artifact/io.github.weblegacy/struts-core/1.5.0-RC1).

Greetings
Stefan

**Reply by anabright** [reply]
ah, my mistake! because it wasn't on [maven central](https://repo.maven.apache.org/maven2/io/github/weblegacy/struts-core/) I thought it hadn't been published. thanks!
